### PR TITLE
Add contributing information

### PR DESCRIPTION
**Pull request by mfeurer**
@amueller a lot of this is actually copied from scikit-learn. Is there a special template that I need to follow to make this legally correct?

### CODE REVIEW

**Comment by mfeurer** [comment]
@amueller do you have some time to review this one? I think you're most experienced with user engagement.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+3. Swith to the ``develop`` branch:
+
+   ```bash
+   $ git checkout develop
+   ```
+
+3. Create a ``feature`` branch to hold your development changes:
+
+   ```bash
+   $ git checkout -b my-feature
```
Comment (nok) [inline CONTRIBUTING.md]: I prefer using the branch type as suffix to keep the overview, i.e.: `feature/idea` or `feature-idea`.

**Comment by amueller** [comment]
sorry @mfeurer not looking at github notifications right now (have 20000+)
What do you mean by legally correct?

**Comment by nok** [comment]
Legally, because the content of these files are partially copied from the [scikit-learn/scikit-learn](https://github.com/scikit-learn/scikit-learn/) repository ([CONTRIBUTING.md](https://github.com/scikit-learn/scikit-learn/blob/master/CONTRIBUTING.md), [ISSUE_TEMPLATE.md](https://github.com/scikit-learn/scikit-learn/blob/master/ISSUE_TEMPLATE.md), [PULL_REQUEST_TEMPLATE.md](https://github.com/scikit-learn/scikit-learn/blob/master/PULL_REQUEST_TEMPLATE.md)) I guess.

**Comment by mfeurer** [comment]
Yep, those files are somewhat copied from scikit-learn, so to follow the BSD-license, we need to mention that. How about following [this](https://opensource.stackexchange.com/questions/2153/copying-a-file-from-an-open-source-project) proposal?

**Comment by mfeurer** [comment]
Okay, I just added a note about the files in question and added the license of scikit-learn to our license file. Should be fine like this.

**Comment by PGijsbers** [comment]
I can't review anymore since I made the last change, right?

**Pull request merged**